tools/analyze_memory_profile.py

```diff
@@ -35,7 +35,7 @@ def add(self, size: int) -> None:
 
 
 def color(pct: float) -> str:
-    return f"{int((100. - pct) // 10) + 1}"
+    return f"{int((100.0 - pct) // 10) + 1}"
```

Comment:
also not great we're copy/pasting this `:]`

```suggestion
    return f"{((100 - pct) // 10) + 1}"
```